### PR TITLE
feat(design-system): ship the card anatomy parts on the token contract

### PR DESCRIPTION
**Pull request by fx**
## Summary

Ships the card anatomy parts on the token contract — the second task of `docs/changes/0010-design-tokens-and-anatomy.md`. This is the first change to stamp the anatomy classes and `data-*` attributes into real markup, so it establishes the stable selector contract that the LCARS theme later depends on.

No card is migrated onto the anatomy and `GridCard` is not restyled — that is the final task. Every existing card test passes unmodified.

## Parts and their contract names

All under `src/components/anatomy/`, styled by `anatomy.css` (wrapped in `@layer liebe-base`, no `!important`, no colour literals).

| Component | Contract class |
|---|---|
| `IconCircle` | `liebe-icon` |
| `CardMeta` / `CardName` / `CardState` | `liebe-name`, `liebe-state` |
| `Pill` + `PillGroup` | `liebe-pill` |
| `Chip` | `liebe-chip` |
| `CardValue` | `liebe-value` |
| `Sparkline` | `liebe-spark` |

`liebe-card` and `liebe-slider` are left to the shell-restyle and slider tasks. Internal, explicitly non-contract names (`liebe-meta`, `liebe-pill-group`, `-label`, `-icon`, `-dot`, `-canvas`, `-line`, `-area`, `-baseline`, `-unit`, `-number`, `-detail`) are documented as such.

Every part stamps `data-domain`, `data-color`, and `data-active` — the last **only when active**, so themes select on `[data-active]` rather than comparing values. All three go through one shared helper so they cannot drift apart. The theming spec's selector contract records this.

## Domain colour

`data-color` maps to three internal properties in the stylesheet (`--part-color`, `--part-tint`, `--part-text`), one rule per triplet, and the parts read only those. No Radix `color` prop and no Radix scale reference at the point of use — a test asserts no `var(--amber-…)`-style reference exists anywhere in the sheet, and another asserts every triplet resolves through `domainColorTokens()`. `src/theme/tokens.ts` gained a derived `DomainColorName` union so a colour cannot be named that the stylesheet has no rule for.

The one styling exception is the bulb-RGB `hue` prop, which overrides the three properties inline with the tint mixed at the same 20% — a data-driven value, which the change document permits.

## Two decisions worth reviewing

**Labels inside a tinted part take the neutral foreground, not the triplet.** The tint pattern is calibrated for a glyph (3:1); a 12.5px label needs 4.5:1, and on the tint neither the base hue nor step 11 clears it (measured 4.1:1 amber, 3.6:1 orange in light). The surface and glyph carry the hue; the label stays readable. Added as a rule to the design-system spec.

**`src/styles/app.css`'s reset and coarse-pointer touch floor are now layered.** This is the one file outside this task's scope that changed, and it was not optional: an unlayered `* { padding: 0 }` outranks every cascade layer, so it was zeroing the anatomy's padding — chips measured `padding: 0px` in the built workshop before the fix, 12px after. The touch floor likewise stretched the 34px chip past `--liebe-chip-height`, i.e. beat a token.

Moving them costs nothing else: at zero specificity the reset already lost to every other author rule, and origin (not layering) is what keeps it ahead of UA defaults. This is a slice of the theming change's "wrap all baseline CSS" task, landing early out of necessity; the theming spec records it as partly done with the remaining sheets still owed. It will matter to the shell restyle too — `.liebe-card { padding: var(--liebe-card-padding) }` would have been dead for the same reason.

## Touch targets

Both interactive parts meet the spec's "≥44px in at least one dimension" on the inline axis, deliberately: meeting it on the block axis would push them past the height their token states, and a tappable chip would then stand taller than the read-only chip beside it. Measured in Chromium under a coarse pointer, `button.liebe-chip` lays out at 44×34; with `min-inline-size` removed it collapses to 24px. A regression test now pins both the floor and the painted height so a future edit cannot silently shrink either.

## Accessibility

Method: built workshop served locally, every story's `iframe.html` driven in headless Chromium with `axe-core` injected, both appearances.

**35 new anatomy stories: 0 critical, 1 serious** — and that one is not a component defect. `--liebe-c-*-text` pins Radix step 11, which is calibrated against step 1 rather than a tinted ground, so amber (4.497:1), orange (4.398:1) and teal (4.450:1) land just under AA on the light card surface. It reproduces on `main` from the token story and cannot be fixed in a component; tracked in **#197**.

Whole workshop: 202 stories, 43 with serious/critical — the documented baseline (**#191**, **#192**) plus those two token surfaces. No new violation class.

Fixed while building: chip and pill label contrast (3.4:1 inactive, 4.1:1 active) and a focus ring that took the part's hue and missed 3:1 on the light ground — now neutral.

## Testing

- `npm test` — 866 passed, 2 skipped
- `npm run lint`, `npm run typecheck` — clean
- `npm run test:coverage` — 100% lines, branches and functions on every new product file
- `npm run build-storybook`, `npm run build:ha:prod` — both succeed

## Scope

The slider primitive and the shell restyle remain; the change document stays `draft` until the final task lands.

## Summary by CodeRabbit

* **New Features**
  * Added reusable card anatomy components (icon circle, meta/name/state, value, pills, chips, and sparklines) with consistent active/inactive rendering and appearance comparisons.
  * Introduced a standardized selector/token contract for domain color theming, plus hue overrides where applicable.
  * Improved interaction behavior (tappable controls, grouped toggles, click handling) and ensured reduced-motion styling.

* **Documentation**
  * Updated the design-system and theming specs with the extended anatomy/theming selector contract and label-contrast rules.

* **Tests**
  * Added/expanded tests covering contracts, accessibility, styling/color rules, and sparkline rendering edge cases.

### CODE REVIEW

File: src/components/anatomy/Sparkline.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+export function Sparkline({ values = [], label, ...part }: SparklineProps) {
+  const attributes = anatomyPart('liebe-spark', part)
+  const shape = values.length > 1 && values.every(Number.isFinite) ? sparkShape(values) : null
+
+  return (
+    <div
+      {...attributes}
+      data-empty={shape ? undefined : 'true'}
+      {...(label ? { role: 'img', 'aria-label': label } : { 'aria-hidden': true })}
+    >
```
[the remark this answers is not in the document]
Comment (fx) [reply src/components/anatomy/Sparkline.tsx:99]: Fixed. `data-active` is now gated on a drawable `shape`, so an empty or non-finite series always renders neutral: with no series there is nothing for the domain colour to describe, and a card whose history has not arrived was showing a saturated "no data" baseline. The placeholder table in `anatomy.test.tsx` gained the `active: true` + empty/invalid case asserting the neutral result.

File: src/components/anatomy/CardMeta.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+ *
+ * It carries the same `data-*` attributes as every other part, so a theme can
+ * reach it (`.liebe-name[data-domain='lock']`), but it never renders state
+ * colour: the name says what the thing is, the line below says what it is
+ * doing.
+ */
+export function CardName({ children, ...part }: CardNameProps) {
+  return (
+    <div {...anatomyPart('liebe-name', { ...part, active: false, hue: undefined })}>{children}</div>
+  )
```
[the remark this answers is not in the document]
Comment (fx) [reply src/components/anatomy/CardMeta.tsx:41]: Fixed, and enforced more broadly than the call sites. Rather than requiring `domain` on `CardName` alone, it is now required on the shared `AnatomyPartProps` and non-optional in `AnatomyPartAttributes`, so no part can render without it — the contract is enforced once in `anatomyPart()` instead of per component. The two stories that omitted it now pass it.

Deliberately not defaulted, for the reason your finding implies: an omitted prop makes React drop the attribute, but a default would trade a missing attribute for a wrong one — a part claiming a domain it does not belong to is harder to notice than one the type refuses to render. `color` does default, because `default` is a real triplet in the palette, so the attribute is still stamped with something themes can select on.

This mattered more than the two call sites suggest: this is the PR that establishes the selector contract, and a silently missing `data-domain` would not have surfaced until the LCARS theme in change 0013 tried to target names.

File: src/components/anatomy/Pill.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  ...part
+}: PillProps) {
+  return (
+    <button
+      type="button"
+      {...anatomyPart('liebe-pill', part)}
+      aria-pressed={part.active ?? false}
+      aria-label={hideLabel ? label : undefined}
+      disabled={disabled}
+      onClick={(event) => {
```
[the remark this answers is not in the document]
Comment (fx) [reply src/components/anatomy/Pill.tsx:84]: Fixed by making `onClick` required rather than defaulting to disabled. The distinction against `Chip`, which does render a `<span>` when it has nothing to do: a chip has a genuine read-only life — "3 lights on" is a summary whether or not you can tap it — whereas a pill is one option in a `PillGroup` selector, and everything it is made of (`aria-pressed`, `disabled`, the equal-width row) presumes that choosing it does something.

So the same underlying rule, never render an inert control, is kept by refusing the inert case in the type instead of degrading the element. Defaulting to `disabled` was rejected on your own reasoning: it would render a control that looks unavailable when it is merely undecorated. A pill that must not fire right now is explicitly `disabled`, which communicates that to assistive technology too.

File: src/components/anatomy/Sparkline.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+/** Trims the coordinate noise that would otherwise fill the DOM. */
+function round(value: number): number {
+  return Math.round(value * 100) / 100
+}
+
+function sparkShape(values: number[]): SparkShape {
+  const min = Math.min(...values)
+  const max = Math.max(...values)
+  const span = max - min
```
[the remark this answers is not in the document]
Comment (fx) [reply src/components/anatomy/Sparkline.tsx]: Fixed — min and max now come from a single pass instead of `Math.min(...values)`. Worth noting it is insurance rather than a live bug: history is downsampled before it reaches the sparkline, so the argument limit is not reachable today. The loop is cheaper regardless, so there was no reason to keep the spread.
